packages/react-core/src/components/NumberInput/NumberInput.tsx

```diff
@@ -87,6 +87,16 @@ export const NumberInput: React.FunctionComponent<NumberInputProps> = ({
   const keyDownHandler =
     inputProps && inputProps.onKeyDown ? inputProps.onKeyDown : defaultKeyDownHandler({ inputName, onMinus, onPlus });
 
+  const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    event.target.value = Number(event.target.value).toString();
```

Comment:
I would create a local event variable and set it to the event then modify that.
```suggestion
    let ev = event.target.value;
    ev.target.value = Number(event.target.value).toString();
```